src/ILCompiler.Compiler/src/CppCodeGen/ILToCppImporter.cs

```diff
@@ -305,6 +309,7 @@ private void AppendCastIfNecessary(StackValueKind dstType, TypeDesc srcType)
                 Append("(");
                 Append(_writer.GetCppSignatureTypeName(srcType));
```

Comment:
This pattern of _writer.GetCppSignatureTypeName and AddTypeReference crops up a lot - we could probably use a helper method that does both so it's clear when we reference something by its C++ signature during ILToCPPImport, we're also adding a dependency to it.